runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -875,7 +875,7 @@ void chpl_comm_exit(int all, int status) {
 }
 
 void  chpl_comm_put(void* addr, c_nodeid_t node, void* raddr,
-                    int32_t elemSize, int32_t typeIndex, int32_t len,
+                    size_t elemSize, int32_t typeIndex, size_t len,
                     int ln, c_string fn) {
   const int size = elemSize*len;
```

Comment:
Oh, I think you're right.  If size_t is an int64_t and 'int' is 'int32_t' (which it normally is), this will be lossy and continue to exhibit Nikhil's bug, right?

One of the annoying things about this fix is that I can't figure out how to file a (portable) test against it because the array size required to trigger it exceeds the amount of memory I can allocate on my local desktop (say).  And because we have no way to "catch" OOM accesses, it's hard to write a safe test of it.  :P